src/tools/compiletest/src/header/tests.rs

```diff
@@ -57,7 +56,9 @@ fn config() -> Config {
         "--target=x86_64-unknown-linux-gnu",
         "--channel=nightly",
     ];
-    let args = args.iter().map(ToString::to_string).collect();
+    let mut args: Vec<String> = args.iter().map(ToString::to_string).collect();
+    args.push("--rustc-path".to_string());
+    args.push(std::env::var("RUSTC").unwrap_or_else(|_| "rustc".to_string()));
```

Comment:
Can we confirm that this is picking up the right rustc as part of bootstrap? I'm a little worried by the or_else here -- when does that become necessary? It seems like an easy way for some obscure failure to sneak in.

I think given that libtest will eat stderr here anyway, we can probably just eprintln! in that path with a "warning: using global rustc" or something like that?

Reply:
Yea, that is a little subtle, I should have left a comment here.  The `RUSTC` here is from stage0. The fallback is only if you are running `cargo test` manually.

The exact version of `rustc` isn't too important here, as the tests that depend on running `rustc` aren't depending on anything too bleeding edge.  If any of those targets get removed (or change properties), then test failures probably won't show up until the beta bump after those changes happen. I apologize ahead of time for when that happens since it usually falls on your shoulders.  I tried to make sure the asserts included extra context information in case it does.

Printing a warning sounds good, though.

Reply:
I know we've run into some problems around recently introduced targets getting linted by check-cfg in stage0 std; I think we'll be ok for tests though since we typically introduce at tier 3 so there's not any tests running in our CI.